model/commitqueue/db.go

```diff
@@ -109,9 +109,13 @@ func remove(id, issue string) error {
 	)
 }
 
+// if the DB queue isn't in the processing state we expect, we should error
 func setProcessing(id string, status bool) error {
 	return updateOne(
-		bson.M{IdKey: id},
+		bson.M{
+			IdKey:         id,
+			ProcessingKey: !status,
```

Comment:
[here](https://github.com/evergreen-ci/evergreen/blob/e1bcd1e3d76bb23d8240bb5f1444b2736303c8fc/model/commitqueue/commit_queue.go#L127) if the head of the queue isn't currently being processed it will return an error. Tracing up the stack, it will return [here](https://github.com/evergreen-ci/evergreen/blob/e1bcd1e3d76bb23d8240bb5f1444b2736303c8fc/service/api_task.go#L234) and not finish `EndTask`